dev_roadmap.md

```diff
@@ -0,0 +1,329 @@
+
+ - QuTiP family packages. The main qutip package already has sub-packages, which are maintained in the main qutip repo. Any packages maitained by the QuTiP organisation will be called QuTiP 'family' packages. Sub-packages within qutip main will be called 'integrated' sub-packages. Some packages will be maintained in their own repos and installed separately within the main qutip folder structure to provide backwards compatibility, these are (will be) called qutip optional sub-packages. Others will be installed in their own folders, but (most likely) have qutip as a dependency -- these will just be called 'family' packages.
+ - QuTiP affilliated packages. Other packages have been developed by others outside of the QuTiP organisation that work with, and are complementary to, qutip. The plan is to give some recognition to those that we deserve worthy of such [this needs clarification]. These packages will not be maintained by the QuTiP Team.
+ 
+ &#x1F537; Do we really need optional subpackages? It seems that are a bit fiddly and as we are side-stepping bw compat with a major version, we could just make QIP a separate package.
+ 
+ ### Family packages
+ #### qutip main
+ *current package status*: family package `qutip`
+ 
+ *planned package status*: family package `qutip`
+ 
+ The in-scope components of the main qutip package all currently reside in the base folder.
+ The plan is to move some components into integrated subpackages as follows.
+ 
+ - `core` quantum objects and operations
+ - `solver` quantum dynamics solvers
+ 
+ What will remain in the base folder will be miscellaneous modules. There may be some opportunity for grouping some into a `visualisation` subpackage. 
+ There is also some potential for renaming, as some module names have underscores, which is unconventional.
+ 
+  #### Qtrl
+ *current package status*: integrated sub-package `qutip.control`
+ 
+ *planned package status*: family package `qtrl`
+ 
+ There are many OSS Python packages for quantum control optimisation. There are also many different algorithms. 
+ The current `control` integrated subpackage provides the GRAPE and CRAB algorithms. 
+ It is too ambitious for QuTiP to attempt (or want) to provide for all options.
+ Control optimisation has been deemed out of scope and hence these components will be separated out into a family package called Qtrl.
+ 
+ Potentially Qtrl may be replaced by separate packages for GRAPE and CRAB, based on the QuTiP Control Framework.
+ 
+ #### QIP
+ *current package status*: integrated sub-package `qutip.qip`
+ 
+ *planned package status*: optional sub-package `qutip.qip`
+ 
+&#x1F537; Is it really necessary for this to be a sub-package? It could just be a separate package.
+
+The QIP subpackage has been deemed out of scope (feature-wise). It also depends on `qutip.control` and hence would be out of scope for dependency reasons.
+A separate repository has already been made for qutip-qip.
+ 
+ #### qutip-symbolic
```

Comment:
This is interesting. Actually, I'm doing some symbolic computing right now. Maybe I can help in fixing some bugs.

Reply:
Cool. There is quite a lot of potential for symbolic stuff inside of qutip.qutip I think. I have been discussing with @hodgestar too, maybe you two should meet to discuss more

Reply:
@BoxiLi I would love to have a quick catch up about what you're working on and see if we can combine forces or work nicely in parallel.